src/core/server/server.api.md

```diff
@@ -2321,6 +2323,16 @@ export interface SavedObjectsCreateOptions extends SavedObjectsBaseOptions {
     version?: string;
 }
 
+// Warning: (ae-forgotten-export) The symbol "SavedObjectsPointInTimeFinderDependencies" needs to be exported by the entry point index.d.ts
```

Comment:
What did I do wrong that caused API extractor to generate these warnings? The referenced type is not meant to be public and is marked `@internal` -- am I forced to export it when it is used in another type which _is_ public?

Reply:
You're exporting the whole `PointInTimeFinder` class I think, and the constructor is part of the class contract, so `SavedObjectsPointInTimeFinderDependencies` which is used by the constructor needs to be exported. Common workaround is to export an interface instead.

Reply:
You're exporting the whole `PointInTimeFinder` class I think, and the constructor is part of the class contract, so `SavedObjectsPointInTimeFinderDependencies` which is used by the constructor needs to be exported. Common workaround is to export an interface instead.

Reply:
That worked! Thanks for the tip.

Reply:
This is the problem line:
```
export declare type SavedObjectsCreatePointInTimeFinderDependencies = Omit<SavedObjectsPointInTimeFinderDependencies, 'logger'>;
```

api-extractor doesn't compile typescript, so it cannot create the final public type from the private type. That's also why there's no useful docs for `SavedObjectsCreatePointInTimeFinderDependencies`. Not 100% sure, but I think even if you export the private type the docs still won't be useful. I wonder what the next docs system does in this case?

Either way, I think splitting these into two separate interfaces (or having the one inherit from the other) would be better from a docs standpoint (and probably also when cmd+clicking through the types to see their definitions). As per @pgayvallet 's other comment, maybe introducing the factory again solves this problem?

Reply:
> splitting these into two separate interfaces (or having the one inherit from the other) would be better from a docs standpoint

That did the trick, docs look much more useful now.